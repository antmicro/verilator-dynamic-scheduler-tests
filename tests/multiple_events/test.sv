module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   event eventA;
   event eventB;
   event eventC;
   event cont;

   initial begin
     @cont;
     $write("Triggering event A!\n");
     ->eventA;
     @cont;
     $write("Triggering event B!\n");
     ->eventB;
     @cont;
     $write("Triggering event C!\n");
     ->eventC;
   end

   initial begin
     for (int i = 0; i < 3; i++) begin
        $write("Waiting for event A, B, or C...\n");
        ->cont;
        @(eventA, eventB, eventC);
        $write("Got the event!\n");
     end
     $write("*-* All Finished *-*\n");
     $finish;
   end
endmodule
