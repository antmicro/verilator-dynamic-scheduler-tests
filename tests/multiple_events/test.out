Waiting for event A, B, or C...
Triggering event A!
Got the event!
Waiting for event A, B, or C...
Triggering event B!
Got the event!
Waiting for event A, B, or C...
Triggering event C!
Got the event!
*-* All Finished *-*
